### PR TITLE
fix: reject partition join if partition has no active members

### PR DESCRIPTION
**Pull request by lenaschoenburg**
This adds validation to the `PartitionJoinApplier` to verify that the partition has at least one active member. If there is none, the partition does not exist and we can't join.

Based on https://github.com/camunda/zeebe/pull/14956 to test the error response.
Related to https://github.com/camunda/zeebe/issues/14905

### CODE REVIEW

File: topology/src/main/java/io/camunda/zeebe/topology/api/TopologyRequestServer.java [hunk 1]
```diff
@@ -122,8 +122,14 @@ private CompletableFuture<Either<ErrorResponse, TopologyChangeResponse>> mapResp
...
-        .thenApply(Either::<ErrorResponse, TopologyChangeResponse>right)
-        .exceptionally(this::mapError);
+        .handle(
+            (response, throwable) -> {
+              if (throwable == null) {
+                return Either.right(response);
+              } else {
+                return mapError(throwable);
+              }
+            });
```
Comment (deepthidevaki) [inline topology/src/main/java/io/camunda/zeebe/topology/api/TopologyRequestServer.java:132]: :upside_down_face: I didn't know you had fixed this. I pushed a fix to the other branch.

**Review by deepthidevaki — APPROVED** [review]
:rocket:

**Comment by deepthidevaki** [comment]
@oleschoenburg I approved it fast. There are a bunch of tests failing. But nothing looks serious, it is probably just that those tests were ignoring this new rejection scenario. Please fix them before merging.

**Comment by deepthidevaki** [comment]
The test run was stuck in `ClusterEndpointIT`. I added a timeout in it, which should fail the test and collect the logs. So we can figure out where it is stuck.

**Comment by deepthidevaki** [comment]
:sob: Didn't help.

**Comment by oleschoenburg** [comment]
@deepthidevaki I think I fixed it in my last two commits :shrug:

**Comment by deepthidevaki** [comment]
So it was just due to concurrent execution?

**Comment by oleschoenburg** [comment]
Yeah I think so :-/

**Comment by oleschoenburg** [comment]
bors merge

**Pull request merged**